src/clr-angular/forms/checkbox/checkbox-container.ts

```diff
@@ -4,23 +4,120 @@
 
 @Component({
   selector: 'clr-checkbox-container',
   template: `
-        <!-- We want the checkbox input to be before the label, always -->
-        <ng-content select="[clrCheckbox]"></ng-content>
-        <ng-content></ng-content>
-        <label *ngIf="_dynamic"></label>
-    `,
-  host: { '[class.checkbox]': 'true' },
-  providers: [ControlIdService],
+    <ng-content select="label"></ng-content>
+    <label *ngIf="!label && addGrid()"></label>
+    <div class="clr-control-container" [class.clr-control-inline]="clrInline" [ngClass]="controlClass()">
+      <ng-content select="clr-checkbox-wrapper"></ng-content>
+      <div class="clr-subtext-wrapper">
+        <ng-content select="clr-control-helper" *ngIf="!invalid"></ng-content>
+        <clr-icon *ngIf="invalid" class="clr-validate-icon" shape="exclamation-circle" aria-hidden="true"></clr-icon>
+        <ng-content select="clr-control-error" *ngIf="invalid"></ng-content>
+      </div>
+    </div>
+  `,
+  host: {
+    '[class.clr-form-control]': 'true',
+    '[class.clr-form-control-disabled]': 'control?.disabled',
+    '[class.clr-row]': 'addGrid()',
+  },
+  providers: [NgControlService, ControlClassService, IfErrorService],
 })
-export class ClrCheckboxContainer implements DynamicWrapper {
-  // Indicates whether the container is dynamically created by the checkbox input itself
-  _dynamic = false;
+export class ClrCheckboxContainer implements OnDestroy {
+  private subscriptions: Subscription[] = [];
+  invalid = false;
+  @ContentChild(ClrLabel) label: ClrLabel;
+  private inline = false;
+  control: NgControl;
+  // private formGroup: AbstractControl;
+
+  /*
+   * Here we want to support the following cases
+   * clrInline - true by presence
+   * clrInline="true|false" - unless it is explicitly false, strings are considered true
+   * [clrInline]="true|false" - expect a boolean
+   */
+  @Input()
+  set clrInline(value: boolean | string) {
+    if (typeof value === 'string') {
+      this.inline = value === 'false' ? false : true;
+    } else {
+      this.inline = !!value;
+    }
+  }
+  get clrInline() {
+    return this.inline;
+  }
+
+  // @TODO Solve for group validation, which doesn't work now with ngModelGroup
+  // Blocked by https://github.com/angular/angular/issues/20268
+  // @Input()
+  // set clrFormGroup(value: FormGroup) {
+  //   this.formGroup = value;
+  // }
+
+  // @Input()
+  // set clrFormArray(value: FormArray) {
+  //   this.formGroup = value;
+  // }
+
+  constructor(
+    private ifErrorService: IfErrorService,
+    @Optional() private layoutService: LayoutService,
+    private controlClassService: ControlClassService,
+    private ngControlService: NgControlService
+  ) {
+    this.subscriptions.push(
+      this.ngControlService.controlChanges.subscribe(control => {
+        this.control = control;
+      })
+    );
+  }
+
+  ngOnInit() {
+    // @TODO put a solution in for form group validation
```

Comment:
Does this still need to be done?

Reply:
This is part of the future work we need to do when we support checkbox group validation.